### PR TITLE
Correct Gibbs link

### LINKED ISSUE #1171: Broken link on  https://turing.ml/dev/docs/using-turing/dynamichmc
Steps to reproduce:

1. I start here: https://turing.ml/
2. Which redirects automatically to here: https://turing.ml/dev/
3. I click on “Documentation” at the top, which brings me to here: https://turing.ml/dev/docs/using-turing/
4. In the left-hand bar, I click on "Using DynamicHMC", which brings me to here: https://turing.ml/dev/docs/using-turing/dynamichmc
5. In the second paragraph, I click on the word "Gibbs", which brings me to this page, which is a 404: https://turing.ml/docs/library/#-turinggibbs--type

### PR DESCRIPTION
**Pull request by cpfiffer**
Fixes #1171.

### CODE REVIEW

File: docs/src/using-turing/dynamichmc.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ title: Using DynamicHMC
 Turing supports the use of [DynamicHMC](https://github.com/tpapp/DynamicHMC.jl) as a sampler through the `DynamicNUTS` function.
 
 
-`DynamicNUTS` is not appropriate for use in compositional inference. If you intend to use [Gibbs](http://turing.ml/docs/library/#-turinggibbs--type) sampling, you must use Turing's native `NUTS` function.
+`DynamicNUTS` is not appropriate for use in compositional inference. If you intend to use [Gibbs](https://turing.ml/dev/docs/library/#Turing.Inference.Gibbs) sampling, you must use Turing's native `NUTS` function.
```
Comment (yebai) [inline docs/src/using-turing/dynamichmc.md]: Maybe a relative link here? 

```suggestion
`DynamicNUTS` is not appropriate for use in compositional inference. If you intend to use [Gibbs](docs/library/#Turing.Inference.Gibbs) sampling, you must use Turing's native `NUTS` function.
```

Comment (cpfiffer) [inline docs/src/using-turing/dynamichmc.md]:
```suggestion
`DynamicNUTS` is not appropriate for use in compositional inference. If you intend to use [Gibbs]({{site.baseurl}}/docs/library/#Turing.Inference.Gibbs) sampling, you must use Turing's native `NUTS` function.
```

Comment (cpfiffer) [reply docs/src/using-turing/dynamichmc.md]: I changed your suggestion to handle the different site URLs.

Comment (yebai) [reply docs/src/using-turing/dynamichmc.md]: thanks, the [link](https://turing.ml/docs/library/#Turing.Inference.Gibbs) seems broken without `dev`, due to #1157.

Comment (yebai) [reply docs/src/using-turing/dynamichmc.md]: maybe add the `dev` for now, until #1157 is fixed. apologies for the noise.

Comment (yebai) [reply docs/src/using-turing/dynamichmc.md]: Actually, is there a way to add version number automatically after `{{site.baseurl}}`?

Comment (cpfiffer) [reply docs/src/using-turing/dynamichmc.md]: `site.baseurl` handles all the versioning, including `dev`.

Comment (yebai) [reply docs/src/using-turing/dynamichmc.md]: many thanks!

**Pull request merged**